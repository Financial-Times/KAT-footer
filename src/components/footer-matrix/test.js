import React from 'react';
import { mount, shallow } from 'enzyme';
import FooterMatrix from './index.js';
import MatrixGroup from './../matrix-group';

describe('FooterMatrix', () => {
  const matrixTestData = {
    matrix: [
      {
        "title": "Test",
        "items": [
          { "text": "test", "href": "test" },
          { "text": "test", "href": "test" }
        ]
      },
      {
        "title": "Test",
        "items": [
          { "text": "test", "href": "test" },
          { "text": "test", "href": "test" }
        ]
      }
    ]
  };

  const nullMatrixTestData = {
    matrix: []
  };

  describe('component', () => {
    const wrapper = mount(<FooterMatrix matrix={matrixTestData.matrix}/>);

    const content =
    <nav className="o-footer__matrix" role="navigation" aria-label="Useful links">
      <MatrixGroup groupData={matrixTestData.matrix[0]} key={0} index={0}/>
      <MatrixGroup groupData={matrixTestData.matrix[1]} key={1} index={1}/>
    </nav>;

    test('has the right content', () => {
      expect(wrapper.contains(content)).toEqual(true);
    });
  });

  test('matches snapshot', () => {
    const wrapper = shallow(<FooterMatrix matrix={matrixTestData.matrix}/>);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  describe('return null if there is no links', () => {
    const wrapper = mount(<FooterMatrix matrix={nullMatrixTestData.marix} />);
    const content =
    <nav className="o-footer__matrix" role="navigation" aria-label="Useful links">
      {null}
    </nav>;

    console.log(wrapper.html());

    test('has the right content', () => {
      expect(wrapper.contains(content)).toEqual(true);
    });
  });
});
