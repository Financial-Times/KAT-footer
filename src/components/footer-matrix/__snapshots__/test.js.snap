// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterMatrix matches snapshot 1`] = `
"<nav className=\\"o-footer__matrix\\" role=\\"navigation\\" aria-label=\\"Useful links\\">
  <MatrixGroup groupData={{...}} index={0} />
  <MatrixGroup groupData={{...}} index={1} />
</nav>"
`;
